src/engine/controls/cuecontrol.cpp

```diff
@@ -868,11 +1038,49 @@ void CueControl::hotcuePositionChanged(HotcueControl* pControl, double newPositi
         if (newPosition == Cue::kNoPosition) {
             detachCue(pControl);
         } else if (newPosition > 0 && newPosition < m_pTrackSamples->get()) {
+            //TODO: Handle Loops
             pCue->setStartPosition(newPosition);
         }
     }
 }
 
+void CueControl::hotcueLengthChanged(HotcueControl* pControl, double newLength) {
+    QMutexLocker lock(&m_mutex);
+    if (!m_pLoadedTrack)
+        return;
+
+    CuePointer pCue(pControl->getCue());
+    if (pCue) {
+        // Setting the length to Cue::kNoPosition is the same as calling hotcue_x_clear
+        if (newLength == Cue::kNoPosition) {
+            pCue->setHotCue(-1);
+            detachCue(pControl);
+        } else {
+            if (newLength >= 0) {
+                pCue->setEndPosition(pCue->getPosition() + newLength);
+            }
+        }
+    }
+}
+
+void CueControl::hotcueTypeChanged(HotcueControl* pControl, double newType) {
+    QMutexLocker lock(&m_mutex);
+    if (!m_pLoadedTrack)
+        return;
+
+    CuePointer pCue(pControl->getCue());
+    if (pCue) {
+        // Setting the type to 0 or -1 is the same as calling hotcue_x_clear
+        int iType = static_cast<int>(newType);
+        if (iType <= 0 || iType > static_cast<int>(mixxx::CueType::AudibleSound)) {
+            pCue->setHotCue(-1);
+            detachCue(pControl);
+        } else {
+            pCue->setType(static_cast<mixxx::CueType>(iType));
```

Comment:
If the type is changed to a loop, should we use beatloop_size to determine the length?

Reply:
Hmm, apparently I don't need that code anyway. I removed it for now and made the CO read-only because otherwise this would need a lot more checks.

Reply:
okay